extensions/amp-iframe/0.1/test/test-amp-iframe.js

```diff
@@ -43,6 +43,23 @@ describe('amp-iframe', () => {
     }
   };
 
+  function waitForJsInIframe() {
+    return new Promise((resolve, reject) => {
```

Comment:
Generally

- event > polling
- polling > random timeout that is hopefully big enough